### PR TITLE
Part 2: More modifications for evpn_multisite

### PR DESCRIPTION
**Pull request by rahushen**
After adding the provider code for puppet, a few alterations needed to be made from a design perspective.

### CODE REVIEW

File: lib/cisco_node_utils/evpn_multisite.rb [hunk 1]
```diff
@@ -34,17 +34,31 @@ def initialize(multisiteid, instantiate=true)
     end
 
     def create
-      Feature.nv_overlay_enable
+      # There is an issue with EX platform using
+      # Feature.nv_overlay_enable generating error code '400'
+      # Using config_set for enabling nv overlay
+      config_set('feature', 'nv_overlay', state: '')
+      sleep 1
```
Comment (mikewiebe) [inline lib/cisco_node_utils/evpn_multisite.rb:41]: Why do we need the sleep?

File: lib/cisco_node_utils/evpn_multisite.rb [hunk 2]
```diff
@@ -34,17 +34,31 @@ def initialize(multisiteid, instantiate=true)
...
       @set_args[:state] = ''
       config_set('evpn_multisite', 'multisite', @set_args)
     end
 
     def destroy
       @set_args[:state] = 'no'
+      # HACK: set time to a dummy value
```
Comment (mikewiebe) [inline lib/cisco_node_utils/evpn_multisite.rb:48]: What are we working around in this case?

File: lib/cisco_node_utils/evpn_stormcontrol.rb [hunk 3]
```diff
@@ -55,16 +55,19 @@ def destroy
...
-
-    def self.multicast
-      config_get('evpn_stormcontrol', 'multicast')
+    # Create a hash of all stormcontrol instances
+    def self.stormcontrol
+      hash = {}
+      type_list = %w(broadcast multicast unicast)
+      type_list.each do |type|
+        level = config_get('evpn_stormcontrol', type)
+        hash[type] = EvpnStormcontrol.new(type, level, false) if level
```
Comment (mikewiebe) [inline lib/cisco_node_utils/evpn_stormcontrol.rb:64]: Based on the documentation on the puppet side, `level` is a property, not a parameter. If it's needed to uniquely identify the resource it should be a parameter for puppet.

File: lib/cisco_node_utils/evpn_multisite.rb (hunk 2 above)
Comment (rahushen) [reply lib/cisco_node_utils/evpn_multisite.rb:48]: for configuring evpn_multisite, the command is as follows:

evpn multisite border-gateway 50.

To unconfigure, the command 'no evpn multisite border-gateway' doesn't work. It needs a multisite id which can be any integer. Ideally, the user while writing the maniest needn't have a multisite id while uncofiguring (ensure => absent). Hence, the hack.

File: lib/cisco_node_utils/evpn_stormcontrol.rb (hunk 3 above)
Comment (rahushen) [reply lib/cisco_node_utils/evpn_stormcontrol.rb:64]: As discussed, the level is not required to uniquely identify the resource.

File: lib/cisco_node_utils/evpn_multisite.rb (hunk 1 above)
Comment (rahushen) [reply lib/cisco_node_utils/evpn_multisite.rb:41]: I just used the logic for self.nv_overlay_enable in feature.rb. I think this is to allow the device to actually turn the feature on after processing the configuration and avoid timing issues.

**Comment by rahushen** [comment]
@mikewiebe ... Thanks for the first round of reviews. I've made changes and the code is now ready for another round of review.

File: lib/cisco_node_utils/cmd_ref/bgp_neighbor_af.yaml [hunk 4]
```diff
@@ -146,11 +146,11 @@ prefix_list_out:
   set_value: '<state> prefix-list <str> out'
   default_value: ''
 
-rewrite_rt_asn:
+rewrite_evpn_rt_asn:
   _exclude: [ios_xr, N3k, N3k-F, N5k, N6k, N7k, N9k-F]
   kind: boolean
-  get_value: '/^rewrite-rt-asn$/'
-  set_value: '<state> rewrite-rt-asn'
+  get_value: '/^rewrite-evpn-rt-asn$/'
```
Comment (mikewiebe) [inline lib/cisco_node_utils/cmd_ref/bgp_neighbor_af.yaml:152]: Does this work on both I7.1 and I7.2?

Comment (rahushen) [reply lib/cisco_node_utils/cmd_ref/bgp_neighbor_af.yaml:152]: Yes, it works across both releases.

**Comment by mikewiebe** [comment]
👍

**Pull request merged**